tst/com/amazon/corretto/crypto/provider/test/LibCryptoRngTest.java

```diff
@@ -0,0 +1,147 @@
+    // zero gaps. To detect this, we'll generate arrays of different lengths
+    // and check to see if certain bytes are always zero
+    @Test
+    public void testNextBytes() {
+        for (int size = 0; size < 64; size++) {
+            final byte[] checkArr = new byte[size];
+            final byte[] arr = new byte[size];
+            for (int trial = 0; trial < 4; trial++) {
+                rnd.nextBytes(arr);
+                System.out.println(Hex.encodeHex(arr));
+                for (int x = 0; x < size; x++) {
+                    checkArr[x] = (byte) (checkArr[x] | arr[x]);
+                }
+            }
+            for (int x = 0; x < size; x++) {
+                assertTrue(0 != checkArr[x],
+                        "Check array size " + size + " position " + x + " is equal to zero");
+            }
+        }
+    }
+
+    // A common mistake it when filling arrays to not do it properly and leave
+    // zero gaps. To detect this, we'll generate arrays of different lengths
+    // and check to see if certain bytes are always zero
+    @Test
+    public void testGenerateSeed() {
+        for (int size = 0; size < 64; size++) {
+            final byte[] checkArr = new byte[size];
+            for (int trial = 0; trial < 4; trial++) {
+                final byte[] arr = rnd.generateSeed(size);
+                System.out.println(Hex.encodeHex(arr));
+                for (int x = 0; x < size; x++) {
+                    checkArr[x] = (byte) (checkArr[x] | arr[x]);
+                }
+            }
+            for (int x = 0; x < size; x++) {
+                assertTrue(0 != checkArr[x],
+                        "Check array size " + size + " position " + x + " is equal to zero");
+            }
+        }
+    }
+
+    // There really isn't a good way to test random numbers.
+    // So we generate a few, ensure they aren't all the same
+    // value and don't throw exceptions
+    @Test
+    public void testInt() {
+        final int initial = rnd.nextInt();
+        for (int trial = 0; trial < 10; trial++) {
+            if (initial != rnd.nextInt()) {
+                return;
+            }
+        }
+        fail("Failed to find a different value");
+    }
+
+    @Test
+    public void testLong() {
+        final long initial = rnd.nextLong();
+        for (int trial = 0; trial < 10; trial++) {
+            if (initial != rnd.nextLong()) {
+                return;
+            }
+        }
+        fail("Failed to find a different value");
+    }
+
+    @Test
+    public void reseed() {
+        // Just ensure this doesn't crash
+        rnd.setSeed(new byte[0]);
+        rnd.setSeed(new byte[1]);
+        rnd.setSeed(new byte[16]);
+        rnd.setSeed(new byte[20]);
+        rnd.setSeed(new byte[24]);
+        rnd.setSeed(new byte[32]);
+        rnd.setSeed(new byte[48]);
+        rnd.setSeed(new byte[64]);
+    }
+
+    @Test
+    public void largeRequest() {
```

Comment:
Assuming you want to test re-seeding in the underlying DRBG.

This is unlikely to cause a re-seed in AWS-LC. There are two relevant numbers:
1) max number of `generate` calls between reseeds, and
2) max number of bytes per `generate` call.

If (1) is reached, AWS-LC will re-seed the CTR-DRBG. For (2), if you request output larger than the maximum, AWS-LC will split the output generation into several calls to `generate`.

(1):
* non-fips `2^12`
* fips `2^24`
* https://github.com/awslabs/aws-lc/blob/main/crypto/fipsmodule/rand/rand.c#L62.

(2):
* `2^16` bytes https://github.com/awslabs/aws-lc/blob/main/crypto/fipsmodule/rand/internal.h#L86
* Splitting logic https://github.com/awslabs/aws-lc/blob/main/crypto/fipsmodule/rand/rand.c#L421

To exercise the different code-paths, assuming AWS-LC and CTR-DRBG, can modify test to:
* Re-seed: `x * 2^12` calls to `RAND_bytes()`, for some scaling factor `x`.
* Splitting: input size requests (strictly) larger than `2^16` bytes.

Reply:
The latter doesn't seem relevant though, since you already have your own splitting logic for much smaller requests
```
private static final int MAX_SINGLE_REQUEST = 8192;
```